src/wp-includes/class-wp-query.php

```diff
@@ -3571,7 +3571,15 @@ public function the_post() {
 		global $post;
 
 		if ( ! $this->in_the_loop ) {
-			update_post_author_caches( $this->posts );
+			// Only prime the post cache for queries limited to the ID field.
+			$post_ids = array_filter( $this->posts, 'is_numeric' );
+			// Exclude any falsey values, such as 0.
+			$post_ids = array_filter( $post_ids );
+			if ( $post_ids ) {
+				_prime_post_caches( $post_ids, $this->query_vars['update_post_term_cache'], $this->query_vars['update_post_meta_cache'] );
+			}
+			$post_objects = array_map( 'get_post', $this->posts );
+			update_post_author_caches( $post_objects );
```

Comment:
It occurs to me, this will have a similar problem to #[56952](https://core.trac.wordpress.org/ticket/56952) and throw an error if called before pluggable functions are loaded.

It might be worth returning early in `update_post_author_caches()` if `cache_users()` is undefined.

Reply:
Agreed.